central/main.go

```diff
@@ -284,11 +285,33 @@ func main() {
 	pkgMetrics.NewDefaultHTTPServer(pkgMetrics.CentralSubsystem).RunForever()
 	pkgMetrics.GatherThrottleMetricsForever(pkgMetrics.CentralSubsystem.String())
 
+	if env.PrivateDiagnosticsEnabled.BooleanSetting() {
+		privateServer := private.NewHTTPServer()
+		privateServer.AddRoutes(privateRoutes())
+		privateServer.RunForever()
+	}
+
 	go startGRPCServer()
 
 	waitForTerminationSignal()
 }
 
+func privateRoutes() []*private.Route {
+	result := make([]*private.Route, 0)
+	for r, h := range routes.DebugRoutes {
+		result = append(result, &private.Route{
+			Route:         r,
+			ServerHandler: h,
+			Compression:   true,
```

Comment:
Let's use `r.Compression` here so we don't run into any trouble in the future in case `routes.DebugRoutes` are being changed.